app/data/tasks/deduplicate_records.py

```diff
@@ -0,0 +1,185 @@
+    This should allow multiple tasks to run at the same time without identifying
+    duplicates more than once.
+
+    Job failure:
+    On job failure, the task should be retried after deleting the duplicates
+    associated with the task.  This is not automatic at this time.
+    A job may fail in a way that the job's failed state is not updated to be ERROR
+    if the celery worker itself goes down.  Otherwise, the task status should be
+    set properly
+    """
+    time_extent = get_time_extent()
+    ids, queryset = get_dedupe_set(time_extent)
+    # this can be adjusted depending on the memory required
+    paginator = Paginator(ids, 1000)
+    job = DedupeJob(status=DedupeJob.Status.STARTED)
+    job.save()
+    starttime = datetime.datetime.now()
+    duplicatecount = 0
+    try:
+        logger.info("Finding duplicates. 1000 records/batch, %d batches total" %
+                    (paginator.num_pages))
+        for page_num in paginator.page_range:
+            page = paginator.page(page_num)
+            records = queryset.in_bulk(page.object_list)
+            done = []
+            for record in records:
+                done.append(record)
+                duplicates = find_duplicates_for_record(record)
+                recordDuplicates = []
+                for duplicate in duplicates:
+                    if duplicate.uuid not in done:
+                        # handle this in the find_duplicates_for_record call
+                        # to reduce number of queries
+                        num_reversed = RecordDuplicate.objects.filter(
+                            record__uuid=duplicate.uuid,
+                            duplicate_record__uuid=record
+                        ).count()
+                        if num_reversed == 0:
+                            rec = Record.objects.get(uuid=record)
+                            recordDuplicates.append(
+                                RecordDuplicate(
+                                    record=rec,
+                                    duplicate_record=duplicate,
+                                    job=job,
+                                    score=calculate_score(rec, duplicate)
+                                )
+                            )
+                            duplicatecount += 1
+                RecordDuplicate.objects.bulk_create(recordDuplicates)
+            logger.info(
+                "Batch %d of %d done. %d duplicates found, %s elapsed" % (
+                    page_num, paginator.num_pages, duplicatecount,
+                    datetime.datetime.now() - starttime
+                )
+            )
+        logger.info("Finished searching for duplicate records.")
+        job.status = DedupeJob.Status.SUCCESS
+    except Exception as e:
+        # TODO: retry job next time it runs with same start time
+        job.status = DedupeJob.Status.ERROR
+        logger.error("Failed: exception=\n%s" % (e, ))
```

Comment:
I'd also add a line here to set the Celery status to failure as well so that it lines up with the status in `job`. http://docs.celeryproject.org/en/latest/reference/celery.app.task.html#celery.app.task.Task.update_state